drivers/rtc/rtc_mc146818.c

```diff
@@ -0,0 +1,592 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT motorola_mc146818_rtc
```

Comment:
Same here, i.e. `motorola_mc146818`, without the `_rtc`

Reply:
motorola_mc146818 is already in use for counter_cmos.c counter. So, added rtc.